File: CSharpCore/HamaraBasketTests/SellByUpdaterTests.cs
using HamaraBasketQualityAssurance;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace HamaraBasketTests
{
    [TestClass]
    public class SellByUpdaterTests
    {
        [TestMethod]
        public void SellByShouldDecreaseByOne()
        {
            var item = new HamaraBasketItem(10, 10, ItemType.Expiring);
            var checker = new SellByUpdater();

            checker.UpdateSellBy(item);

            Assert.AreEqual(9, item.SellBy);
        }
    }
}

File: Java/src/test/java/com/hamaraBasket/SellByUpdaterTests.java
package com.hamaraBasket;

import static org.junit.Assert.assertEquals;

import org.junit.Test;

public class SellByUpdaterTests
    {
        @Test
        public void sellByShouldDecreaseByOne()
        {
            HamaraBasketItem item = new HamaraBasketItem(10, 10, ItemType.Expiring);
            SellByUpdater checker = new SellByUpdater();

            checker.updateSellBy(item);

            assertEquals(9, item.getSellBy());
        }
    }